### ISSUE TITLE
BUG: When the OCR of a file is not complete, Qiqqa does not give control to the user 

### ISSUE THREAD

**Issue by raindropsfromsky** [jimmejardine/qiqqa-open-source#166]
I have a pdf file that has machine-readable (searchable) text (as opposed to scanned images).

I just noticed a subtle sign in the left-hand-top corner of one page. When I hovered my mouse on it, I learned that-
![image](https://user-images.githubusercontent.com/9047168/77431835-446a5d00-6e03-11ea-9d3b-8ac76ce96d3d.png)

Well, I would NOT like to wait: I would like to know _where else_ this problem exists, and how to solve it as quickly as possible. (The file has large fonts with high contrast, so even if Qiqqa wants to OCR a searchable text, I don't mind.)

But Qiqqa does not give me a chance. I expect- 
1. The status line should show a statistics ("x out of y pages need OCR")
2. A summary figure of how many pages have the same issue, followed by a clickable page-list.
3. A control to OCR all the pending pages at once
4. A control on each "defective" page to re-OCR the page.

The file in which I noticed the issue is attached:
[SC judgement dtd 17-03-2020, on EIA for PRR, Bangalore.pdf](https://github.com/jimmejardine/qiqqa-open-source/files/4375586/SC.judgement.dtd.17-03-2020.on.EIA.for.PRR.Bangalore.pdf)

**Comment by GerHobbelt** [comment]
Related to #35 et al.

**Comment by GerHobbelt** [comment]
See also: https://github.com/jimmejardine/qiqqa-open-source/issues/165#issuecomment-603402113-permalink

>
> But Qiqqa does not give me a chance. I expect- 
> 1. The status line should show a statistics ("x out of y pages need OCR")
> 2. A summary figure of how many pages have the same issue, followed by a clickable page-list.
> 3. A control to OCR all the pending pages at once
> 4. A control on each "defective" page to re-OCR the page.

My current thought on this is:

Qiqqa SHOULD indeed have a way to better inform its users what's going on under the hood. Given the plethora of  possibilities where and how things can go wrong, using an already severely abused status line is not going to cut it. Given https://github.com/jimmejardine/qiqqa-open-source/issues/138, plus  my own experience and now this, this starts to sound more like a 'IDE', where one 'develops' his/hers complex document collection. (Think: status/log panel where status info is reported. When needed, detailed info can be provided to the user there.)

Though at first glance I like the wish list, it certainly implies a quite significant development effort as this app will then turn into a PDF/document OCR processing & management UI. I do  recall my struggles with those OmniPage products in the past  **way too vividly** to believe I'ld be able to produce something like that, and *working*, going on a one-man show, which Qiqqa currently is. 

Thus there's either funding and a team or some hard choices to make. *Oh wait.* 🤡  There's always some hard choices there as the *funders* will want... 😉

Anyway, YMMV.

**Comment by raindropsfromsky** [comment]
Yes, where several parameters are changing, and all have to be reported simultaneously to the user, it would need a **console panel**, rather than a single status line, which cannot handle multiple values.

I have seen status lines that show multiple parameters and their values. 
But that's an ugly design.

**Comment by raindropsfromsky** [comment]
Rather than shelving this change, why not work on a simpler design?

In my suggested scenario, we don't need a console or status line; and that is because the status is not required to be displayed all the time. Just make it on-demand.

Here's the workflow:
The user hovers his mouse on the ![image](https://user-images.githubusercontent.com/9047168/77464727-9f18ae80-6e2d-11ea-8006-1c737d2ad89a.png) icon.

Qiqqa shows a pop-up dialog, with the statistics:

Pages to be texified:  n
Pages to be OCRed:   m
Time to completion:  hh:mm

Below that, a single button to let the user decide whether he wants to make the thread a foreground process. That means the process will consume a major part of system resources but will finish much faster. 

But let the user also decide to interrupt and make the process a background process with low priority. 

So the single button can toggle between **Turbo** mode and **Background** mode.

**Comment by Atif-Anwer** [comment]
Agreed with the comments above; there should be a clear (with popup) status bar that can give information about the current PDF as well as the processes that are being run in queue (right now its just a linear status bar overlaying new tasks on top of each other). Can incorporate in the design easily too. 
I need to add a UI task board somewhere... or maybe use the exisitng Github Projects/Kanban board for the UI feature list only.
